src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let subSeparator = separator;
+
+    if (separator === null) {
+      subSeparator = 'null';
+    }
+
+    if (this.length === 0) {
+      return '';
```

Comment:
```suggestion
      return result';
```
result - це пустий рядок початково. Можеш повертати його